src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -33,15 +40,23 @@ namespace syrec {
         for (const auto& [input, output]: tt) {
             // truth table has to be completely specified
             assert(input[0].has_value());
-            assert(output[0].has_value());
-            const auto       in    = *input[0];
-            const auto       out   = *output[0];
-            const auto       index = static_cast<std::size_t>(out) * 2U + static_cast<std::size_t>(in);
-            TruthTable::Cube reducedInput(input.begin() + 1, input.end());
-            TruthTable::Cube reducedOutput(output.begin() + 1, output.end());
-            subTables.at(index).try_emplace(std::move(reducedInput), std::move(reducedOutput));
-        }
 
+            const auto in  = *input[0];
+            const auto out = *output[0];
+
+            if (output[0].has_value()) {
+                const auto       index = static_cast<std::size_t>(out) * 2U + static_cast<std::size_t>(in);
+                TruthTable::Cube reducedInput(input.begin() + 1, input.end());
+                TruthTable::Cube reducedOutput(output.begin() + 1, output.end());
```

Comment:
These lines are duplicated in the else branch. Consider moving them outside of the `if`.